lang/test/org/partiql/lang/eval/EvaluatingCompilerDateTimeTests.kt

```diff
@@ -34,4 +63,187 @@ class EvaluatingCompilerDateTimeTests : EvaluatorTestBase() {
+     */
+    private class ArgumentsForTimeLiterals : ArgumentsProviderBase() {
+        private val LOCAL_TIMEZONE_OFFSET = ZoneOffset.systemDefault().rules.getOffset(Instant.now())
+        private val LOCAL_TZ_MINUTES = LOCAL_TIMEZONE_OFFSET.totalSeconds / 60
+
+        private fun case(query: String, expected: String, expectedTime: Time? = null) = TimeTestCase(query, expected, expectedTime)
+
+        override fun getParameters() = listOf(
+            case("TIME '00:00:00.000'", "00:00:00.000000000", Time(0, 0, 0)),
+            case("TIME '23:59:59.99999999'", "23:59:59.999999990", Time(23, 59, 59, 999999990)),
+            case("TIME (2) '23:59:59.99999999'", "00:00:00.000000000", Time(0, 0, 0)),
+            case("TIME '00:45:13.840800524'", "00:45:13.840800524", Time(0, 45, 13, 840800524)),
+            case("TIME '05:20:52.015779149'", "05:20:52.015779149", Time(5, 20, 52, 15779149)),
+            case("TIME '23:59:59'", "23:59:59.000000000", Time(23, 59, 59)),
+            case("TIME (12) '12:24:12.123'", "12:24:12.123000000", Time(12, 24, 12, 123000000)),
+            case("TIME (0) '12:59:59.9'", "13:00:00.000000000", Time(13, 0,0)),
+            case("TIME WITH TIME ZONE '00:00:00'", "00:00:00.000000000${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", Time(0,0,0,0,null, LOCAL_TZ_MINUTES)),
+            case("TIME (2) WITH TIME ZONE '12:24:12.123'", "12:24:12.120000000${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", Time(12, 24, 12, 120000000, 2, LOCAL_TZ_MINUTES)),
+            case("TIME (2) WITH TIME ZONE '12:24:12.123-00:00'", "12:24:12.120000000+00:00", Time(12, 24, 12, 120000000, 2, 0)),
+            case("TIME (2) WITH TIME ZONE '12:24:12.123+00:00'", "12:24:12.120000000+00:00", Time(12, 24, 12, 120000000, 2, 0)),
+            case("TIME (2) WITH TIME ZONE '12:24:12.123+05:30'", "12:24:12.120000000+05:30", Time(12, 24, 12, 120000000, 2, 330)),
+            case("TIME (2) WITH TIME ZONE '12:59:59.135-05:30'", "12:59:59.140000000-05:30", Time(12, 59, 59, 140000000, 2, -330)),
+            case("TIME (2) WITH TIME ZONE '12:59:59.134-05:30'", "12:59:59.130000000-05:30", Time(12, 59, 59, 130000000, 2, -330)),
+            case("TIME '12:25:12.123456' IS TIME", "true"),
+            case("TIME (2) '01:01:12' IS TIME", "true"),
+            case("TIME WITH TIME ZONE '12:25:12.123456' IS TIME", "true"),
+            case("TIME (2) WITH TIME ZONE '01:01:12' IS TIME", "true"),
+            case("'01:01:12' IS TIME", "false")
+        )
+    }
+
+    private val RANDOM_GENERATOR = generateRandomSeed()
+
+    private fun generateRandomSeed() : Random {
+        val seed = Random(1).nextInt()
+        println("Randomly generated seed is ${seed}. Use this to reproduce failures in dev environment.")
+        return Random(seed)
+    }
+
+    data class Time(val hour: Int, val minute: Int, val second: Int, val nano: Int = 0, val precision: Int? = null, val tz_minutes: Int? = null) {
```

Comment:
The fact that you have this implies that a wrapper type in the library might be appropriate, but I know I sound like a broken record.

Reply:
I have included the wrapper class. On an orthogonal note, the class is only for the test purpose. I have renamed it to `TimeForTest` for more clarity.